lib/src/rules/unnecessary_const.dart

```diff
@@ -82,6 +82,8 @@ class _Visitor extends SimpleAstVisitor {
   void _visitTypedLiteral(TypedLiteral node) {
     if (node.constKeyword == null) return;
 
+    // todo (pq): update to use linter context API
```

Comment:
I think you should be able to ask `node.inConstantContext`.

Reply:
Yes, it looks that it will work.
One complication is that it declared in `ExpressionImpl`.

Reply:
Should we make it public? Do we need to wrap it in `LintContext`?

Reply:
We don't need manipulations with AST to implement it, so I think we can make it public.

Do you like its name? Should it be `isInConstantContext`? OTOH we also have `inGetterContext`, so it seems consistent. Just don't know if consistently good, or consistently bad :-)

Reply:
Thanks!  Seems like this could go in `LinterContext`?

Reply:
(whoops.  missed your earlier comment.)

> We don't need manipulations with AST to implement it, so I think we can make it public.

I can't speak to consistency.  Either work for me.  I guess I'd lean towards `inConstantContext`.  If nothing else, it's shorter!